File: src/EntertainmentApi/AntiClown.Entertainment.Api.Dto/F1Predictions/Statistics/F1SeasonStatsDto.cs
namespace AntiClown.Entertainment.Api.Dto.F1Predictions.Statistics;

public class F1SeasonStatsDto
{
    public DriverStatisticsDto[] TenthPlacePointsRating { get; set; }
    public DriverStatisticsDto[] MostPickedForTenthPlace { get; set; }
    public DriverStatisticsDto[] TenthPickedButDnfed { get; set; }
    public DriverStatisticsDto[] DriverOwnTenthPlacePoints { get; set; }
    public DriverStatisticsDto[] TenthPlacePredictionEfficiency { get; set; }
    public DriverStatisticsDto[] MostDnfDrivers { get; set; }
    public DriverStatisticsDto[] MostPickedForDnf { get; set; }
    public UserStatisticsDto[] TenthPlaceDnfAntiRating { get; set; }
    public LeadGapPredictionStatsDto[] ClosestLeadGapPredictions { get; set; }
    public DriverStatisticsDto[] SafetyCarPickCounts { get; set; }
    public DriverStatisticsDto[] SafetyCarActualCounts { get; set; }
    public DriverStatisticsDto[] SafetyCarCorrectGuesses { get; set; }
}

File: src/Web/AntiClown.Web.Front/src/Dto/F1Predictions/F1SeasonStatsDto.ts
import { DriverStatisticsDto } from "./DriverStatisticsDto";
import { UserStatisticsDto } from "./UserStatisticsDto";
import { LeadGapPredictionStatsDto } from "./LeadGapPredictionStatsDto";

export interface F1SeasonStatsDto {
  tenthPlacePointsRating: DriverStatisticsDto[];
  mostPickedForTenthPlace: DriverStatisticsDto[];
  tenthPickedButDnfed: DriverStatisticsDto[];
  driverOwnTenthPlacePoints: DriverStatisticsDto[];
  tenthPlacePredictionEfficiency: DriverStatisticsDto[];
  mostDnfDrivers: DriverStatisticsDto[];
  mostPickedForDnf: DriverStatisticsDto[];
  tenthPlaceDnfAntiRating: UserStatisticsDto[];
  closestLeadGapPredictions: LeadGapPredictionStatsDto[];
  safetyCarPickCounts: DriverStatisticsDto[];
  safetyCarActualCounts: DriverStatisticsDto[];
  safetyCarCorrectGuesses: DriverStatisticsDto[];
}
